File: src/main/kotlin/ru/nsu/fit/neofr/model/ResourceEntity.java
package ru.nsu.fit.neofr.model;

import javax.persistence.*;

@Entity
@Table(name = "Resource", schema = "dbo", catalog = "HabitCounter")
public class ResourceEntity {
    private int resourceId;
    private String resourceName;
    private String resourceDescription;
    private Double totalCount;

    @Id
    @Column(name = "resource_id")
    public int getResourceId() {
        return resourceId;
    }

    public void setResourceId(int resourceId) {
        this.resourceId = resourceId;
    }

    @Basic
    @Column(name = "resource_name")
    public String getResourceName() {
        return resourceName;
    }

    public void setResourceName(String resourceName) {
        this.resourceName = resourceName;
    }

    @Basic
    @Column(name = "resource_description")
    public String getResourceDescription() {
        return resourceDescription;
    }

    public void setResourceDescription(String resourceDescription) {
        this.resourceDescription = resourceDescription;
    }

    @Basic
    @Column(name = "total_count")
    public Double getTotalCount() {
        return totalCount;
    }

    public void setTotalCount(Double totalCount) {
        this.totalCount = totalCount;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ResourceEntity that = (ResourceEntity) o;

        if (resourceId != that.resourceId) return false;
        if (resourceName != null ? !resourceName.equals(that.resourceName) : that.resourceName != null) return false;
        if (resourceDescription != null ? !resourceDescription.equals(that.resourceDescription) : that.resourceDescription != null)
            return false;
        if (totalCount != null ? !totalCount.equals(that.totalCount) : that.totalCount != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = resourceId;
        result = 31 * result + (resourceName != null ? resourceName.hashCode() : 0);
        result = 31 * result + (resourceDescription != null ? resourceDescription.hashCode() : 0);
        result = 31 * result + (totalCount != null ? totalCount.hashCode() : 0);
        return result;
    }
}

File: src/main/kotlin/ru/nsu/fit/neofr/Model/ResourceEntity.kt
package ru.nsu.fit.neofr.Model

import javax.persistence.*

@Entity
@Table(name = "Resource", schema = "dbo", catalog = "HabitCounter")
class ResourceEntity {
    @get:Id
    @get:Column(name = "resource_id")
    var resourceId: Int = 0
    @get:Basic
    @get:Column(name = "resource_name")
    var resourceName: String? = null
    @get:Basic
    @get:Column(name = "resource_description")
    var resourceDescription: String? = null
    @get:Basic
    @get:Column(name = "total_count")
    var totalCount: Double? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val that = o as ResourceEntity?

        if (resourceId != that!!.resourceId) return false
        if (if (resourceName != null) resourceName != that.resourceName else that.resourceName != null) return false
        if (if (resourceDescription != null) resourceDescription != that.resourceDescription else that.resourceDescription != null)
            return false
        return if (if (totalCount != null) totalCount != that.totalCount else that.totalCount != null) false else true

    }

    override fun hashCode(): Int {
        var result = resourceId
        result = 31 * result + if (resourceName != null) resourceName!!.hashCode() else 0
        result = 31 * result + if (resourceDescription != null) resourceDescription!!.hashCode() else 0
        result = 31 * result + if (totalCount != null) totalCount!!.hashCode() else 0
        return result
    }
}
